src/lib.rs

```diff
@@ -11,28 +11,214 @@ use rand::Rng;
 
+
+#[derive(PartialEq, Eq, Debug)]
+pub struct Grid {
+    grid: CellSet,
+    bounds: Option<(usize, usize)>,
+}
+
+impl Grid {
+    fn new(b: Option<(usize, usize)>) -> Self {
+        match b {
+            None => Grid {
+                grid: CellSet::new(),
+                bounds: b,
+            },
+            _ => Grid {
+                grid: CellSet::with_capacity(b.unwrap().0 * b.unwrap().1),
+                bounds: b,
+            },
+        }
+
+    }
+    fn insert(&mut self, cell: Cell) {
+        if self.bounds != None {
+            if cell.0 <= self.bounds.unwrap().0 && cell.1 <= self.bounds.unwrap().1 {
+                self.grid.insert(cell);
+            }
+        } else {
+            self.grid.insert(cell);
+        }
+    }
+    fn contains(&self, cell: &Cell) -> bool {
+        self.grid.contains(cell)
+    }
+
+    fn x_bound(&self) -> usize {
+        self.bounds.unwrap().0
+    }
+
+    fn y_bound(&self) -> usize {
+        self.bounds.unwrap().1
+    }
+
+    pub fn gen(&mut self) {
+        if self.bounds == None {
+            return;
+        }
+        self.grid.clear();
+        for x in 0..self.bounds.unwrap().0 {
+            for y in 0..self.bounds.unwrap().1 {
+                if rand::thread_rng().gen_weighted_bool(10) {
+                    self.insert((x, y));
+                }
+            }
+        }
+    }
+}
+
+impl From<Vec<String>> for Grid {
+    /// Returns a Grid interpreted from a string representation
+    ///
+    /// # Arguments
+    ///
+    /// * `s` - Representation of the grid. Each element of the vector
+    /// represents a row in the grid. Hash marks # indicate live cells.
+    /// For example ovec!["#  ", "   ", " # "] represents a grid with live
+    /// cells at (0, 0) and (2, 1).
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # #[macro_use] extern crate hemoglobin; fn main(){
+    ///
+    /// let grid = hemoglobin::Grid::from(ovec!["#  ", "   ", " # "]);
+    /// # }
+    /// ```
+    fn from(s: Vec<String>) -> Self {
+        let mut result = Grid::new(None);
+        for (y, row) in s.iter().enumerate() {
+            for (x, c) in row.chars().enumerate() {
+                if c == '#' {
+                    result.insert((x, y));
+                }
+            }
+        }
+        result
+    }
+}
+
+pub struct Rule {
+    dec: BigInt,
+    bin: BitVec,
+}
+
+impl From<BigInt> for Rule {
+    fn from(x: BigInt) -> Self {
+        Rule {
+            dec: x.clone(),
+            bin: {
+                let result_reversed = BitVec::from_bytes(&x.to_bytes_be().1);
+                let mut result = BitVec::from_elem(512, false);
+                for i in 0..result_reversed.len() {
+                    result.set(i, result_reversed[result_reversed.len() - i - 1]);
+                }
+                result
+            },
+        }
+    }
+}
+
+impl From<String> for Rule {
+    fn from(s: String) -> Self {
+        Rule::from(s.parse::<BigInt>().unwrap())
```

Comment:
Having `unwrap` in a conversion like this seems bad since it might panic. Should probably implement a conversion to a `Result` type that could be an error instead (like parsing to `BigInt` itself).

Reply:
This is true, and I argued about this with Daniel the other day about whether or not we should change some of our return types to `Result` (take the `x_bound()` function for example), he disagreed on some instances, so I feel like we should move this into it's own issue to discuss where we should and shouldn't use `Result`